tests/h/services/search_index/_queue_test.py

```diff
@@ -137,206 +136,192 @@ def add_where(self, queue):
 
 class TestSync:
     def test_it_does_nothing_if_the_queue_is_empty(self, batch_indexer, queue):
-        counts = queue.sync(LIMIT)
+        counts = queue.sync(1)
 
         assert counts == {}
         batch_indexer.index.assert_not_called()
 
     def test_it_ignores_jobs_that_arent_scheduled_yet(
-        self, add_all, batch_indexer, queue
+        self, batch_indexer, now, queue, SyncAnnotationJobFactory
     ):
-        add_all(schedule_in=ONE_WEEK_IN_SECONDS)
+        SyncAnnotationJobFactory(scheduled_at=now + datetime_.timedelta(hours=1))
 
-        counts = queue.sync(LIMIT)
+        counts = queue.sync(1)
 
         assert counts == {}
         batch_indexer.index.assert_not_called()
 
-    @pytest.mark.usefixtures("with_queued_annotations")
     def test_it_ignores_jobs_beyond_limit(
-        self, all_annotation_ids, batch_indexer, queue
+        self, batch_indexer, queue, SyncAnnotationJobFactory
     ):
-        queue.sync(LIMIT)
+        limit = 1
+        SyncAnnotationJobFactory.create_batch(size=limit + 1)
+
+        queue.sync(limit)
 
-        for annotation_id in all_annotation_ids[LIMIT:]:
-            assert annotation_id not in batch_indexer.index.call_args[0][0]
+        assert len(batch_indexer.index.call_args[0][0]) == limit
```

Comment:
This test previously asserted that all the annotations beyond `LIMIT` were _not_ in `batch_indexer.index.call_args`:

```python
for annotation_id in all_annotation_ids[LIMIT:]:
    assert annotation_id not in batch_indexer.index.call_args[0][0]
```

But this was flakey: only `LIMIT` annotations would ever be processed but exactly _which_ `LIMIT` annotations gets processed varies. This is because, when the fixture adds jobs to the queue for all `LIMIT` annotations`, two jobs can end up with the exact same `enqueued_at` time, and then their sort order is random!

I could actually reliably reproduce this: run the test multiple times, sometimes it'd fail.

It's enough to assert that only `limit` annotations were indexed. This test doesn't care which ones.